battle-engine.js

```diff
@@ -514,6 +510,12 @@ BattlePokemon = (function () {
 		}
 		return null;
 	};
+	BattlePokemon.prototype.ignoreAbility = function () {
```

Comment:
This sounds like a function to start ignoring an ability, honestly. You might want to rename it `isSuppressingAbility` or something.